### PR TITLE
ticdc: add document for ticdc support data-dir.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

add `data-dir` description in TiCDC deployment.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/deploy-ticdc.md [hunk 1]
```diff
@@ -61,4 +61,5 @@ cdc server --pd=http://10.0.10.25:2379 --log-file=ticdc_3.log --addr=0.0.0.0:830
 - `cert`：TiCDC 使用的证书文件路径，PEM 格式，可选。
 - `key`：TiCDC 使用的证书密钥文件路径，PEM 格式，可选。
 - `config`：可选项，表示 TiCDC 使用的配置文件地址。TiCDC 从 v5.0.0 开始支持该选项，TiUP 从 v1.4.0 开始支持在部署 TiCDC 时使用该配置。
-- `sort-dir`：指定排序引擎使用的临时文件目录。该配置项的默认值为 `/tmp/cdc_sort`。在开启 Unified Sorter 的情况下，如果服务器的该目录不可写或可用空间不足，请手动指定 `sort-dir`。需确保 TiCDC 在该 `sort-dir` 路径下可读写数据。
+- `sort-dir`：指定排序引擎使用的临时文件目录。该配置项的默认值为 `/tmp/sorter`。在开启 Unified Sorter 的情况下，如果服务器的该目录不可写或可用空间不足，请手动指定 `sort-dir`。需确保 TiCDC 在该 `sort-dir` 路径下可读写数据。从 v4.0.14 和 v5.0.3 (除 v5.1.0-alpha) 开始，该参数将不起作用，推荐使用 `data-dir` 参数启动 TiCDC。
```
Comment (liuzix) [inline ticdc/deploy-ticdc.md:64]: `v5.1.0-alpha` is not a released version

File: ticdc/deploy-ticdc.md [hunk 2]
```diff
@@ -61,4 +61,5 @@ cdc server --pd=http://10.0.10.25:2379 --log-file=ticdc_3.log --addr=0.0.0.0:830
...
+- `data-dir`：指定 TiCDC 运行过程中使用的临时文件目录，从 v4.0.14 和 v5.0.3 (除 v5.1.0-alpha) 开始支持。推荐在启动 TiCDC 时指定该参数，建议其所在磁盘至少有 500GB 的可用空间。如果启动时未指定该参数，将会参考已有 changefeed 的 `sort-dir` 设置情况初始化该参数，默认值为 `/tmp/cdc_data`。在运行过程中，如果 `data-dir` 所在磁盘可用磁盘空间不足 10%, 运行在该 TiCDC server 上的 changefeed 将会报错并停止运行，需要用户在重新设置好 `data-dir` 所在磁盘情况之后手动重启该相关 changefeed。`sort-dir` 将会被挂载为其子目录，固定为 `{data-dir}/tmp/sorter`。
```
Comment (liuzix) [inline ticdc/deploy-ticdc.md:65]: `挂载为其子目录` is not a good way of saying this.

**Comment by TomShawn** [comment]
@JinLingChristopher @asddongmen @amyangfei Please help confirm whether this PR applies to v5.1. Thanks~

File: ticdc/deploy-ticdc.md [hunk 3]
```diff
@@ -61,4 +61,5 @@ cdc server --pd=http://10.0.10.25:2379 --log-file=ticdc_3.log --addr=0.0.0.0:830
 - `cert`：TiCDC 使用的证书文件路径，PEM 格式，可选。
 - `key`：TiCDC 使用的证书密钥文件路径，PEM 格式，可选。
 - `config`：可选项，表示 TiCDC 使用的配置文件地址。TiCDC 从 v5.0.0 开始支持该选项，TiUP 从 v1.4.0 开始支持在部署 TiCDC 时使用该配置。
-- `sort-dir`：指定排序引擎使用的临时文件目录。该配置项的默认值为 `/tmp/cdc_sort`。在开启 Unified Sorter 的情况下，如果服务器的该目录不可写或可用空间不足，请手动指定 `sort-dir`。需确保 TiCDC 在该 `sort-dir` 路径下可读写数据。
+- `sort-dir`：指定排序引擎使用的临时文件目录。该配置项的默认值为 `/tmp/sorter`。在开启 Unified Sorter 的情况下，如果服务器的该目录不可写或可用空间不足，请手动指定 `sort-dir`。需确保 TiCDC 在该 `sort-dir` 路径下可读写数据。从 v4.0.14 和 v5.0.3 开始，该参数将不起作用，推荐使用 `data-dir` 参数启动 TiCDC。
+- `data-dir`：指定 TiCDC 运行过程中使用的临时文件目录，从 v4.0.14 和 v5.0.3 开始支持。推荐在启动 TiCDC 时指定该参数，建议其所在磁盘至少有 500GB 的可用空间。如果启动时未指定该参数，将会参考已有 changefeed 的 `sort-dir` 设置情况初始化该参数，默认值为 `/tmp/cdc_data`。在运行过程中，如果 `data-dir` 所在磁盘可用磁盘空间不足 10%, 运行在该 TiCDC server 上的 changefeed 将会报错并停止运行，用户需要重新设置 `data-dir` 所在磁盘, 在至少保证有 10% 可用空间之后，手动重启相关 changefeed。排序引擎使用的临时文件目录，即 `sort-dir`，会被设置为 `{data-dir}/tmp/sorter`。
```
Comment (amyangfei) [inline ticdc/deploy-ticdc.md:65]: 是从 v5.0.2 开始支持 `data-dir`? @3AceShowHand

Comment (3AceShowHand) [reply ticdc/deploy-ticdc.md:65]: > 是从 v5.0.2 开始支持 `data-dir`? @3AceShowHand

v4.0.14 和 v5.0.3 开始支持

Comment (TomShawn) [reply ticdc/deploy-ticdc.md:65]: 这里和 https://github.com/pingcap/docs-cn/pull/6687/files#diff-33150822cf42e37369ebf2cd8dec35142aefda117ef5015af5755e6465d56d02R383 这个 PR 不一致。6687 里说的是 v5.0.2 开始支持的。能否再确定一下吗？

Comment (3AceShowHand) [reply ticdc/deploy-ticdc.md:65]: > 这里和 https://github.com/pingcap/docs-cn/pull/6687/files#diff-33150822cf42e37369ebf2cd8dec35142aefda117ef5015af5755e6465d56d02R383 这个 PR 不一致。6687 里说的是 v5.0.2 开始支持的。能否再确定一下吗？

TiUP 是  v5.0.2 支持，TiCDC 是 v5.0.3 支持，是没有问题的。

Comment (TomShawn) [reply ticdc/deploy-ticdc.md:65]: 用户在集群实际操作中，能使用到 data-dir 的版本是哪一个版本呢？

Comment (TomShawn) [reply ticdc/deploy-ticdc.md:65]: 参考这里 https://docs.pingcap.com/zh/tidb/dev/ticdc-overview#sort-dir-%E5%8F%8A-data-dir-%E9%85%8D%E7%BD%AE%E9%A1%B9%E7%9A%84%E5%85%BC%E5%AE%B9%E6%80%A7%E8%AF%B4%E6%98%8E
```suggestion
- `data-dir`：指定 TiCDC 运行过程中使用的临时文件目录，从 v4.0.14 和 v5.0.2 开始支持。推荐在启动 TiCDC 时指定该参数，建议其所在磁盘至少有 500GB 的可用空间。如果启动时未指定该参数，将会参考已有 changefeed 的 `sort-dir` 设置情况初始化该参数，默认值为 `/tmp/cdc_data`。在运行过程中，如果 `data-dir` 所在磁盘可用磁盘空间不足 10%, 运行在该 TiCDC server 上的 changefeed 将会报错并停止运行，用户需要重新设置 `data-dir` 所在磁盘, 在至少保证有 10% 可用空间之后，手动重启相关 changefeed。排序引擎使用的临时文件目录，即 `sort-dir`，会被设置为 `{data-dir}/tmp/sorter`。
```

File: ticdc/deploy-ticdc.md [hunk 4]
```diff
@@ -61,4 +61,5 @@ cdc server --pd=http://10.0.10.25:2379 --log-file=ticdc_3.log --addr=0.0.0.0:830
 - `cert`：TiCDC 使用的证书文件路径，PEM 格式，可选。
 - `key`：TiCDC 使用的证书密钥文件路径，PEM 格式，可选。
 - `config`：可选项，表示 TiCDC 使用的配置文件地址。TiCDC 从 v5.0.0 开始支持该选项，TiUP 从 v1.4.0 开始支持在部署 TiCDC 时使用该配置。
-- `sort-dir`：指定排序引擎使用的临时文件目录。该配置项的默认值为 `/tmp/cdc_sort`。在开启 Unified Sorter 的情况下，如果服务器的该目录不可写或可用空间不足，请手动指定 `sort-dir`。需确保 TiCDC 在该 `sort-dir` 路径下可读写数据。
+- `sort-dir`：指定排序引擎使用的临时文件目录。该配置项的默认值为 `/tmp/sorter`。在开启 Unified Sorter 的情况下，如果服务器的该目录不可写或可用空间不足，请手动指定 `sort-dir`。需确保 TiCDC 在该 `sort-dir` 路径下可读写数据。从 v4.0.14 和 v5.0.3 开始，该参数将不起作用，推荐使用 `data-dir` 参数启动 TiCDC。
+- `data-dir`：指定 TiCDC 运行过程中使用的临时文件目录，从 v4.0.14 和 v5.0.2 开始支持。推荐在启动 TiCDC 时指定该参数，建议其所在磁盘至少有 500GB 的可用空间。如果启动时未指定该参数，将会参考已有 changefeed 的 `sort-dir` 设置情况初始化该参数，默认值为 `/tmp/cdc_data`。在运行过程中，如果 `data-dir` 所在磁盘可用磁盘空间不足 10%, 运行在该 TiCDC server 上的 changefeed 将会报错并停止运行，用户需要重新设置 `data-dir` 所在磁盘, 在至少保证有 10% 可用空间之后，手动重启相关 changefeed。排序引擎使用的临时文件目录，即 `sort-dir`，会被设置为 `{data-dir}/tmp/sorter`。
```
Comment (liuzix) [inline ticdc/deploy-ticdc.md:65]: `将会参考已有 changefeed 的 `sort-dir` 设置情况初始化该参数`

Is this feature affected by a bug?

Comment (3AceShowHand) [reply ticdc/deploy-ticdc.md:65]: As far as I know, this works properly at the moment, didn't find any bug yet.

**Comment by 3AceShowHand** [comment]
close by #6789

**Comment by TomShawn** [comment]
This PR is duplicated with https://github.com/pingcap/docs-cn/pull/6789.

**Pull request closed without merging**